public/app/core/components/PermissionList/AddPermission.tsx

```diff
@@ -88,59 +88,57 @@ class AddPermissions extends Component<Props, NewDashboardAclItem> {
     const pickerClassName = 'min-width-20';
     const isValid = this.isValid();
+    const styles = getStyles(config.theme);
+
     return (
-      <div className="gf-form-inline cta-form">
+      <div className="cta-form">
         <button className="cta-form__close btn btn-transparent" onClick={onCancel}>
           <Icon name="times" />
         </button>
-        <form name="addPermission" onSubmit={this.onSubmit}>
-          <h5>Add Permission For</h5>
-          <div className="gf-form-inline">
-            <div className="gf-form">
-              <div className="gf-form-select-wrapper">
-                <select className="gf-form-input gf-size-auto" value={newItem.type} onChange={this.onTypeChanged}>
-                  {dashboardAclTargets.map((option, idx) => {
-                    return (
-                      <option key={idx} value={option.value}>
-                        {option.text}
-                      </option>
-                    );
-                  })}
-                </select>
-              </div>
-            </div>
-
-            {newItem.type === AclTarget.User ? (
-              <div className="gf-form">
+        <h5>Add Permission For</h5>
+        <Form maxWidth="none" onSubmit={this.onSubmit}>
+          {() => (
+            <HorizontalGroup>
+              <Select
+                isSearchable={false}
+                value={this.state.type}
+                options={dashboardAclTargets}
+                onChange={this.onTypeChanged}
+              />
+
+              {newItem.type === AclTarget.User ? (
                 <UserPicker onSelected={this.onUserSelected} className={pickerClassName} />
-              </div>
-            ) : null}
+              ) : null}
 
-            {newItem.type === AclTarget.Team ? (
-              <div className="gf-form">
+              {newItem.type === AclTarget.Team ? (
                 <TeamPicker onSelected={this.onTeamSelected} className={pickerClassName} />
-              </div>
-            ) : null}
+              ) : null}
+
+              <span className={styles.label}>Can</span>
 
-            <div className="gf-form">
               <Select
                 isSearchable={false}
+                value={this.state.permission}
                 options={dashboardPermissionLevels}
                 onChange={this.onPermissionChanged}
-                className="gf-form-select-box__control--menu-right"
+                width={25}
               />
-            </div>
-
-            <div className="gf-form">
-              <button data-save-permission className="btn btn-primary" type="submit" disabled={!isValid}>
+              <Button data-save-permission type="submit" disabled={!isValid}>
                 Save
-              </button>
-            </div>
-          </div>
-        </form>
+              </Button>
+            </HorizontalGroup>
+          )}
+        </Form>
       </div>
     );
   }
 }
 
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
```

Comment:
FYI: you don't need to use styles factory. You can define the style getter and use [`useStyles` hook](https://github.com/grafana/grafana/blob/master/contribute/style-guides/styling.md#styling-with-theme) that will provide the theme automatically and guarantee memoization

Reply:
Maybe I'm misunderstanding but to the best of my knowledge hooks can't be used in class components. Did I miss something? 🤔

Reply:
Nope, it we me missing that this component is a class one. Ignore my comment :)